uniffi_macros/src/lib.rs

```diff
@@ -369,14 +369,12 @@ pub fn generate_and_include_scaffolding(udl_file: TokenStream) -> TokenStream {
     }.into()
 }
 
-/// A dummy macro that does nothing.
-///
-/// This exists so `#[uniffi::export]` can emit its input verbatim without
-/// causing unexpected errors, plus some extra code in case everything is okay.
-///
-/// It is important for `#[uniffi::export]` to not raise unexpected errors if it
-/// fails to parse the input as this happens very often when the proc-macro is
-/// run on an incomplete input by rust-analyzer while the developer is typing.
+/// An attribute for constructors.
+/// Constructors are in `impl` blocks which have a `#[uniffi::export]` attribute.
+/// Because there's no `derive` for the block, it's impossible to have
+/// `#[uniffi(constructor)]`.
+/// This means we can't give errors in many cases where it's used incorrectly, so
+/// it's just a noop in those cases.
```

Comment:
Probably not *that* many, we could put some code here that parses an optional list of extra attributes plus maybe a few tokens afterwards and checks if those tokens are the start of a `fn` or a different item – could error out if the input is a `const` / `static` / type definition rather than a function definition, and maybe even if it's a method (has a `self` argument).